creator-node/test/StateMonitoringManager.test.js

```diff
@@ -0,0 +1,271 @@
+const { getLibsMock } = require('./lib/libsMock')
+
+describe('test StateMonitoringManager initialization, events, and re-enqueuing', function () {
+  let server, sandbox
+  beforeEach(async function () {
+    const appInfo = await getApp(getLibsMock())
+    await appInfo.app.get('redisClient').flushdb()
+    server = appInfo.server
+    sandbox = sinon.createSandbox()
+
+    nock.disableNetConnect()
+  })
+
+  afterEach(async function () {
+    await server.close()
+    nock.cleanAll()
+    nock.enableNetConnect()
+    sandbox.restore()
+  })
+
+  function getProcessJobMock() {
+    const loggerStub = {
+      info: sinon.stub(),
+      warn: sinon.stub(),
+      error: sinon.stub()
+    }
+    const createChildLogger = sinon.stub().returns(loggerStub)
+    const processJobMock = proxyquire(
+      '../src/services/stateMachineManager/processJob.js',
+      {
+        '../../logging': {
+          createChildLogger
+        }
+      }
+    )
+    return { processJobMock, loggerStub }
+  }
+
+  it('creates the queue and registers its event handlers', async function () {
+    // Initialize StateMonitoringManager and spy on its registerQueueEventHandlersAndJobProcessors function
+    const stateMonitoringManager = new StateMonitoringManager()
+    sandbox.spy(
+      stateMonitoringManager,
+      'registerQueueEventHandlersAndJobProcessors'
+    )
+    const queue = await stateMonitoringManager.init('discoveryNodeEndpoint')
+
+    // Verify that the queue was successfully initialized and that its event listeners were registered
+    expect(queue).to.exist.and.to.be.instanceOf(BullQueue)
+    expect(stateMonitoringManager.registerQueueEventHandlersAndJobProcessors).to
+      .have.been.calledOnce
+    expect(
+      stateMonitoringManager.registerQueueEventHandlersAndJobProcessors.getCall(
+        0
+      ).args[0]
+    )
+      .to.have.property('queue')
+      .that.has.deep.property('name', QUEUE_NAMES.STATE_MONITORING)
+  })
+
+  it('kicks off an initial job when initting', async function () {
+    // Mock the latest userId, which is used during init as an upper bound
+    // to start the monitoring queue at a random user
+    const discoveryNodeEndpoint = 'https://discoveryNodeEndpoint.co'
+    nock(discoveryNodeEndpoint).get('/latest/user').reply(200, { data: 0 })
+    config.set('stateMonitoringQueueRateLimitJobsPerInterval', 1)
+    config.set('stateMonitoringQueueRateLimitInterval', 60_000)
+    const MockStateMonitoringManager = proxyquire(
+      '../src/services/stateMachineManager/stateMonitoring/index.js',
+      {
+        '../../../config': config
+      }
+    )
+
+    // Initialize StateMonitoringManager
+    const stateMonitoringManager = new MockStateMonitoringManager()
+    const queue = await stateMonitoringManager.init(discoveryNodeEndpoint)
+
+    // Verify that the queue has the correct initial job in it
+    return expect(queue.getJobs('delayed'))
+      .to.eventually.be.fulfilled.and.have.nested.property('[0]')
```

Comment:
when `.to.eventually.be.fulfilled.and.have.nested` is used, is there a timeout happening here to wait for the promise to be resolved? / how does this chai chaining work?

Reply:
the `.to.eventually.be.fulfilled` is just one thing, and the `.and` chains on the result of the last assertion (so the fulfilled promise in this case). afaik this chai extension (chai-as-promised) doesn't add any additional timeouts, so we rely on the default mocha timeout for the whole test (I think it's 20 or 30 seconds). it just waits for it to be resolved just like it would if you awaited a promise in the test